src/test/java/seedu/address/model/appointment/AppointmentContainsDatePredicateTest.java

```diff
@@ -0,0 +1,123 @@
+package seedu.address.model.appointment;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.commons.util.DateUtil.DATE_FORMATTER;
+import static seedu.address.commons.util.DateUtil.DATE_TIME_FORMATTER;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.person.Person;
+import seedu.address.testutil.PersonBuilder;
+
+public class AppointmentContainsDatePredicateTest {
+
+    @Test
+    public void equals() {
+        LocalDate firstDate = LocalDate.parse("23-10-2024", DATE_FORMATTER);
+        LocalDate secondDate = LocalDate.parse("10-12-2023", DATE_FORMATTER);
+
+        AppointmentContainsDatePredicate firstPredicate = new AppointmentContainsDatePredicate(firstDate);
+        AppointmentContainsDatePredicate secondPredicate = new AppointmentContainsDatePredicate(secondDate);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        AppointmentContainsDatePredicate firstPredicateCopy =
+                new AppointmentContainsDatePredicate(firstDate);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different keywords but same field -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_appointmentContainsDate_returnsTrue() {
```

Comment:
I like your exhaustive test cases which follows the Boundary Value Analysis as well by testing the boundaries.